### ISSUE TITLE
Towne - draft invoices

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1416]
In GitLab by @kdjstudios on Mar 26, 2019, 08:49

**Submitted by:** Deb Crown <dcrown@towneanswering.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/7786254

**Server:** External

**Client/Site:** Towne

**Account:** All

**Issue:** 

Today is our billing day and I am going into some draft invoices to make manual adjustments and something is happening that was not happening last month:

 

When in a draft invoice, I will click on the item I want to adjust and it pops up in a window.

When I highlight it in anticipation of deleting that number and entering a new one, the number DOES highlight, but – as soon as I am finished highlighting…(on the upclick of my mouse)…the pop-up window disappears and I have to start over.

 

Note: When I do the same action s-l-o-w-l-y it allows me to do it.

 

Obviously, this is not a huge deal…it’s just strangely different (in a negative way) and so I thought I’d report it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 26, 2019, 08:52

Deb wrote:

> Lest you think I am the crazy lady from Towne...I am just reporting this because it is different and I find it annoyingly odd but I have already learned to work with it.
> I'm not sure it's worth doing anything about it...and it's very possible that no one else will even notice it.
> 
> Have a great day!!!!!!!!!!!!

I wrote:

> Thank you for notifying us of this. The pop up window should not be disappearing to my knowledge for any reason. We have informed the support team and will follow up with their research.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 26, 2019, 08:58

@smasih @amishra @pchaudhary 

I tested this on the demo site on external production, but can not seem to reproduce the pop up disappearing. May your team please test this too and see if you can reproduce this issue. Also if we are logging any information that might be helpful in determining why that pop up would be disappearing would be great.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 20, 2019, 06:55

@kdjstudios 

We looked into this but We were unable to reproduce the pop-up issue.

If you share with screenshot It would be great.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 20, 2019, 12:46

I wrote:

> We are following up to see if you are still experiencing the pop up disappearing still? Our support team has been looking into the issue and so far has not been able to reproduce it. if you are still experiencing it could you provide the steps you are taking to have this occur so the support team may better find the cause. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 20, 2019, 13:07

Deb wrote:

> Can you please send the original email I sent? I forget exactly how I described it.
> 
>  I think it still happens but if no one else can make it happen, so be it.
> 
> Never mind - I just did it and it still happens.
> 
> What it boils down to is that you can't do it quickly...


I wrote:

Sorry I did not see the first email where you descried the steps. I believe the support team was not highlighting it fast enough as now I am able to reproduce it when I highlight it very quickly. Let me follow up with the support team with this updated information.  

Deb wrote:

> ....and don't bother thinking about doing anything about it. There are more important things to do...

@tim.traylor @amishra 

I would have to agree with Deb, that there are other things we should be looking into before this, but it does have a bit of concern to me that the popup will only disappear when highlighted quickly. It does not appear to have any direct negative effect on anything else other then slowing the user down. 

Updating the priority, as we may not need to look into this now, but I would think at some point we should.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 1, 2020, 08:48

@kdjstudios  @Robleski 

We have tried to recreate the issue but we are not able to recreate it. So please provide the steps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2020, 09:08

@rranjan @Robleski 

I recorded a video since there is a clear outline of the steps above to recreate this. You should be able to see that when I quickly highlight the input field it auto closes without me clicking the accept or the cancel buttons. But when I am slow it works as intended. 

![GMT20200701-150433_Tech-Servi_1920x1080](/uploads/499cd1ff33ca05c3d995f15ac6a91276/GMT20200701-150433_Tech-Servi_1920x1080.mp4)

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 6, 2020, 07:53

@kdjstudios  @Robleski 

Thanks for clarification

We looked into it and following is our finding:- 

*  This is default pop up functionality of javascript that when you move your cursor outside the area of pop up then it will hide. 

*  If you need to change this functionality then we need to change in the javascript editable library itself.

  Please confirm what we need to do in this situation?

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 6, 2020, 14:51

I wrote:

> We are just following up as the support team was able to determine what is causing the pop up to close. 

It actually was not due to speed, but when you move the mouse/cursor while highlighting outside of the popup dialog, the web browser considers that when you release the button it is a click outside of the dialog. So the web browser closes the popup. 

We are researching if there is any options to work around the web browsers restriction. However, our current solution is just to insure that when highlighting the field is to be sure to not move the mouse outside of the popup dialog. 

@rranjan @ranand @amishra @Robleski @aznbyte 

What is we were to get rid of the pop up and just edit it directly on the web-page? Or is there a different dialog/popup that will work better then what is currently there?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 7, 2020, 05:22

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 8, 2020, 05:02

@kdjstudios @Robleski 

Thank you for the update.

I would suggest we can hide the pop only once click on the **OK** or **Cancel**.
If they click outside we will not hide the pop up.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 8, 2020, 10:32

@amishra 

Please post an estimate to do that.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 15, 2020, 09:18

@kdjstudios @Robleski 

* We looked into this and we did not find yet to disable the outside hide pop-up.
* Because, this is using the **EDITABLE JQUERY LIBRARY** we as per my research till today we unable to fo find how we can fix as we suggested above.

* We need more time to research it and update accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 15, 2020, 09:30

Moving this back to wishlist. further development is on hold.
